openbb_platform/providers/fmp/openbb_fmp/models/financial_ratios.py

```diff
@@ -226,37 +237,54 @@ async def aextract_data(
 
         base_url = "https://financialmodelingprep.com/api/v3"
 
-        ttm_url = f"{base_url}/ratios-ttm/{query.symbol}?&apikey={api_key}"
+        ttm_dict = {"period": "TTM", "date": datetime.now().strftime("%Y-%m-%d")}
 
-        url = (
-            f"{base_url}/ratios/{query.symbol}?"
-            f"period={query.period}&limit={query.limit}&apikey={api_key}"
-            if query.period != "ttm"
-            else ttm_url
-        )
-        results = await amake_request(
-            url, response_callback=response_callback, **kwargs
-        )
+        include_ttm = query.period != "ttm" and query.with_ttm
+
+        async def response_callback(
+            response: ClientResponse, session: ClientSession
+        ) -> List[Dict]:
+            results: List[dict] = await response.json()  # type: ignore
+            symbol = response.url.parts[-1]
+
+            # TTM data
+            ttm_url = f"{base_url}/ratios-ttm/{symbol}?&apikey={api_key}"
+            if include_ttm and (ratios_ttm := await session.get_one(ttm_url)):
+                results.insert(
+                    0,
+                    {"symbol": symbol, **ttm_dict, **ratios_ttm},
+                )
+
+            if query.period == "ttm":
+                results = [{"symbol": symbol, **ttm_dict, **item} for item in results]
 
-        if not results:
-            raise EmptyDataError(f"No data found for the symbol {query.symbol}.")
+            return results
 
-        return results  # type: ignore
+        endpoint = "ratios" if query.period != "ttm" else "ratios-ttm"
+
+        urls = [f"{base_url}/{endpoint}/{symbol}" for symbol in query.symbol.split(",")]
+
+        kwargs.update(
+            params={"period": query.period, "limit": query.limit, "apikey": api_key}
+        )
+
+        return await amake_requests(urls, response_callback=response_callback, **kwargs)
 
     @staticmethod
     def transform_data(
         query: FMPFinancialRatiosQueryParams, data: List[Dict], **kwargs: Any
     ) -> List[FMPFinancialRatiosData]:
         """Return the transformed data."""
-        results = [
-            {to_snake_case(k).replace("ttm", ""): v for k, v in item.items()}
-            for item in data
-        ]
-        if query.period == "ttm":
-            results[0].update(
-                {"period": "TTM", "date": datetime.now().date().strftime("%Y-%m-%d")}
-            )
-        for item in results:
-            item.pop("symbol", None)
-            item.pop("dividend_yiel_percentage", None)
-        return [FMPFinancialRatiosData.model_validate(d) for d in results]
+        results: List[FMPFinancialRatiosData] = []
+        for item in data:
+            new_item = {to_snake_case(k).replace("ttm", ""): v for k, v in item.items()}
```

Comment:
![Screenshot 2024-04-19 at 4 18 17 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/5c1479c5-0227-4283-916e-0aff000db748)

Looks like we need to add "_" to the str.replace(), and "dividend_yiel_percentage" is a specialty for the "TTM" response. It's popped because it is redundant.

Reply:
nice yup done!

Reply:
also have no idea why this one keeps failing on the test runs 
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/61340027/c4797f78-cf12-4ffb-a618-52b5c549a911)
lol

Reply:
> nice yup done!

"dividend_yield_percentage" can be dropped. This is redundant.

PE ratio and peg ratio, maybe some others are also being duplicated.

![Screenshot 2024-04-19 at 10 00 58 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/4ac854ac-76cf-49d7-847d-a180e16016be)

Reply:
> also have no idea why this one keeps failing on the test runs 

I think this is because of way the requests are structured - the nested async requests within the callback.